internal/serialization/compact_field.go

```diff
@@ -0,0 +1,47 @@
+package serialization
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hazelcast/hazelcast-go-client/serialization"
+)
+
+type CompactField struct {
+	Name  string
+	Type  CompactFieldType
+	Value any
+}
+
+var compactReaders = map[serialization.FieldKind]compactFieldReader{
+	serialization.FieldKindString: func(r serialization.CompactReader, field string) any {
+		v := r.ReadString(field)
+		if v == nil {
+			return nil
+		}
+		return *v
```

Comment:
Isn't this logic suppose to be inside the `ReadString`, why do we explicitly edit the return? I see the same operation in https://github.com/yuce/hazelcast-go-client/blob/compact-serialization-temp/internal/serialization/default_compact_reader.go#L224.

Reply:
We want to return a `string` here, not `*string`.